miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultWorksetService.java

```diff
@@ -0,0 +1,204 @@
+import uk.ac.bbsrc.tgac.miso.service.security.AuthorizationManager;
+
+@Service
+@Transactional(rollbackFor = Exception.class)
+public class DefaultWorksetService implements WorksetService {
+
+  @Autowired
+  private WorksetStore worksetStore;
+  @Autowired
+  private SampleService sampleService;
+  @Autowired
+  private LibraryService libraryService;
+  @Autowired
+  private LibraryDilutionService dilutionService;
+  @Autowired
+  private DeletionStore deletionStore;
+  @Autowired
+  private AuthorizationManager authorizationManager;
+
+  public void setWorksetStore(WorksetStore worksetStore) {
+    this.worksetStore = worksetStore;
+  }
+
+  public void setSampleService(SampleService sampleService) {
+    this.sampleService = sampleService;
+  }
+
+  public void setLibraryService(LibraryService libraryService) {
+    this.libraryService = libraryService;
+  }
+
+  public void setDilutionService(LibraryDilutionService dilutionService) {
+    this.dilutionService = dilutionService;
+  }
+
+  public void setDeletionStore(DeletionStore deletionStore) {
+    this.deletionStore = deletionStore;
+  }
+
+  public void setAuthorizationManager(AuthorizationManager authorizationManager) {
+    this.authorizationManager = authorizationManager;
+  }
+
+  @Override
+  public long count(Consumer<String> errorHandler, PaginationFilter... filter) throws IOException {
+    return worksetStore.count(errorHandler, filter);
+  }
+
+  @Override
+  public List<Workset> list(Consumer<String> errorHandler, int offset, int limit, boolean sortDir, String sortCol,
+      PaginationFilter... filter) throws IOException {
+    return worksetStore.list(errorHandler, offset, limit, sortDir, sortCol, filter);
+  }
+
+  @Override
+  public Workset get(long id) {
+    return worksetStore.get(id);
+  }
+
+  @Override
+  public List<Workset> listBySearch(String query) {
+    return worksetStore.listBySearch(query);
+  }
+
+  @Override
+  public List<Workset> listBySample(long sampleId) {
+    return worksetStore.listBySample(sampleId);
+  }
+
+  @Override
+  public List<Workset> listByLibrary(long libraryId) {
+    return worksetStore.listBySample(libraryId);
+  }
+
+  @Override
+  public List<Workset> listByDilution(long dilutionId) {
+    return worksetStore.listBySample(dilutionId);
+  }
+
+  @Override
+  public long save(Workset workset) throws IOException {
+    if (!workset.isSaved()) {
+      loadMembers(workset);
+      validateChange(workset, null);
+      workset.setChangeDetails(authorizationManager.getCurrentUser());
+      return worksetStore.save(workset);
+    } else {
+      Workset managed = worksetStore.get(workset.getId());
+      validateChange(workset, managed);
+      applyChanges(workset, managed);
+      managed.setChangeDetails(authorizationManager.getCurrentUser());
+      return worksetStore.save(managed);
+    }
+  }
+
+  private void loadMembers(Workset newWorkflow) throws IOException {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'java.io.IOException', as it cannot be thrown from method's body. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)